src/main/java/io/fabric8/maven/docker/StartMojo.java

```diff
@@ -483,6 +488,45 @@ private String getTcpHost(WaitConfiguration.TcpConfiguration tcpConfig, Properti
         return host;
     }
 
+    private WaitUtil.WaitChecker getHealthyWaitChecker(final String imageConfigDesc, final DockerAccess docker, final String containerId, final List<String> logOut) {
+
+
+        return new WaitUtil.WaitChecker() {
+
+            private boolean first = true;
+
+            @Override
+            public boolean check() {
+                try {
+                    final InspectedContainer container = docker.getContainer(containerId);
+                    if (container == null) {
+                        log.debug("HealthyWaitChecker:  Container %s not found");
+                        return false;
+                    }
+
+                    if (first) {
```

Comment:
Can we be absolutely sure that the container already exists within the docker api when `waitIfRequested` is called? I am not sure if the docker `create container`-api gives us such guarantees (iirc i moved the healthcheck resolution into the check during development because the container could not be fetched once)